Framework/Source/Sample.h

```diff
@@ -161,17 +161,60 @@ namespace Falcor
         float mCurrentTime = 0;                           ///< Global time
         ArgList mArgList;
 
+		//	The Memory Check for one frame.
+		struct MemoryCheck
+		{	
+			unsigned long pFrame;
+			float pTime;
+			float pEffectiveTime;
+			unsigned long long pTotalVirtualMemory;
+			unsigned long long pTotalUsedVirtualMemory;
+			unsigned long long pCurrentlyUsedVirtualMemory;
+		};
+
+		//	The Memory Check Between Frames.
+		struct MemoryCheckRange
+		{
+			bool active = false;
+			MemoryCheck startCheck;
+			MemoryCheck endCheck;
+		};
+
+		//	The List of Memory Check Ranges.
+		MemoryCheckRange memoryFrameCheckRange;
+		MemoryCheckRange memoryTimeCheckRange;
+
+
     protected:
         void renderFrame() override;
         void handleWindowSizeChange() override;
         void handleKeyboardEvent(const KeyboardEvent& keyEvent) override;
         void handleMouseEvent(const MouseEvent& mouseEvent) override;
         virtual float getTimeScale() final { return mTimeScale; }
         void initVideoCapture();
-        void captureScreen();
+	
+		/**	Capture the Current Memory and write it to the provided memory check.
```

Comment:
Why is that part of Sample? Shouldn't it be a utility function is OS.h where the sample test can just call?